### PR TITLE
Test fixes for Jython and a few improvements

### PR DESCRIPTION
**Pull request by ProgVal**
Mainly test fixes. But also:
- Fix to `edge_boundary`: https://github.com/networkx/networkx/commit/5e3da7202a2ea5a7302974172b86c14341961b2e
- MultiGraph support for `assert_edges_equal`: https://github.com/networkx/networkx/commit/bed56aa0b1ec270c4c772f386c97dbd7aa0132b0

### CODE REVIEW

File: networkx/algorithms/boundary.py [hunk 1]
```diff
@@ -89,9 +89,13 @@ def edge_boundary(G, nbunch1, nbunch2=None, data=False, keys=False,
     # implemented by using the `not in` operator, instead of by creating
     # an additional set and using the `in` operator.
     if nbunch2 is None:
-        return (e for e in edges if e[1] not in nset1)
+        return (e for e in edges
```
Comment (jfinkels) [inline networkx/algorithms/boundary.py]: This could be `return (e for e in edges if (e[0] in nset1) ^ (e[1] in nset1))`, using the bitwise xor operator.

Comment (ProgVal) [inline networkx/algorithms/boundary.py]: Indeed. Fixed.

**Comment by jfinkels** [comment]
I think the `edge_boundary` issue is that when I call, for example, `G.edges({0, 1, 2})` I get all edges incident to node 0, node 1, or node 2. The original code assumes that the edges come out in the form `(0, v)`, `(1, v)`, `(2, v)`, etc., but in other Python implementations, the edge may come out as `(v, 1)`, `(v, 2)`, etc. I think this issue is worth a separate pull request.

**Comment by hagberg** [comment]
Thanks, I get the issue with `edge_boundary` now.
We should definitely fix that.

**Comment by hagberg** [comment]
See #2203 #2207 #2209 #2305

**Pull request closed without merging**